libsolidity/formal/SMTEncoder.cpp

```diff
@@ -663,6 +663,17 @@ void SMTEncoder::endVisit(FunctionCall const& _funCall)
 	case FunctionType::Kind::Event:
 		// These can be safely ignored.
 		break;
+	case FunctionType::Kind::ObjectCreation:
+	{
+		ArrayType const& arrayType = dynamic_cast<ArrayType const&>(*_funCall.annotation().type);
+		// TODO: support other kinds of arrays
```

Comment:
I actually think this works for any kind of array, as it just presets them to zero and sets the length.

Reply:
Yes it works for any Solidity array. Maybe you can already change that here?

Reply:
So just remove the warning and consider it working for everything?

Reply:
Is `ObjectCreation` only for arrays? I think so, right? struct constructors go into function call

Reply:
Struct constructors are separate.